corporate/lib/stripe.py

```diff
@@ -1968,14 +2008,20 @@ def do_upgrade(self, upgrade_request: UpgradeRequest) -> dict[str, Any]:
             self.ensure_current_plan_is_upgradable(customer, upgrade_request.tier)
         billing_modality = upgrade_request.billing_modality
         schedule = upgrade_request.schedule
+
         license_management = upgrade_request.license_management
-        licenses = upgrade_request.licenses
+        if billing_modality == "send_invoice":
+            license_management = "manual"
 
-        seat_count = unsign_seat_count(upgrade_request.signed_seat_count, upgrade_request.salt)
+        licenses = upgrade_request.licenses
+        request_seat_count = unsign_seat_count(
```

Comment:
The signed seat count is still checked before we check for any changes we might know about to the current seat count for an organization.